File: Frontend/src/domain/valueObject/title.ts

export class Title {
    private readonly value: string
    constructor(input: string) {
        Title.validation(input);
        this.value = input
    }

    Value() {
        return this.value
    }

    static validation(input: string) {
        if ([...input].length > 50) {
            throw new Error("タイトルは50字以内にしてください")
        }
    }
}

File: GoAPI/src/domain/valueobject/Title.go
package valueobject

import (
	"fmt"
	"unicode/utf8"
)

type Title struct {
	title string
}

const maxtitle = 50

func NewTitle(input string) (Title, error) {
	if utf8.RuneCountInString(input) > maxtitle {
		return Title{}, fmt.Errorf("タイトルは%d字以下にしてください", maxtitle)
	}

	if input == "" {
		return Title{}, fmt.Errorf("タイトルは必須です")
	}

	return Title{title: input}, nil
}

func (t Title) Value() string {
	return t.title
}
